chain-signatures/node/src/mesh/mod.rs

```diff
@@ -32,6 +32,10 @@ pub struct MeshState {
     /// Participants that are active in the network; as in they respond when pinged.
     pub active: Participants,
 
+    /// Participants that are currently out-of-sync, they will become active
+    /// once we finished synchronization.
+    pub need_sync: Vec<Participant>,
```

Comment:
this probably should be a HashSet instead, since `status` is called multiple times where we can possibly push the same participant into this list, and end up syncing multiple times with the same node

Reply:
nevermind about this, I forgot that `status()` returns a new `MeshState` instead of modifying the old one